boa/src/builtins/map/ordered_map.rs

```diff
@@ -94,6 +94,16 @@ where
         self.0.get(key)
     }
 
+    /// Get a key-value pair by index
+    /// Valid indices are 0 <= index < self.len()
+    /// Computes in O(1) time.
+    ///
+    /// TODO perhaps return Record as per
+    /// https://tc39.es/ecma262/#sec-%mapiteratorprototype%.next bullet 12.a.
```

Comment:
```suggestion
    /// Get a key-value pair by index
    ///
    /// Valid indices are `0 <= index < self.len()`
    /// Computes in O(1) time.
```
We do return a record of `(key, value)` so there is no need for the todo

Reply:
Is this Option<(&K, &V)> a Record?

Reply:
> The Record type is used to describe data aggregations within the algorithms of this specification. A Record type value consists of one or more named fields. The value of each field is either an ECMAScript value or an abstract value represented by a name associated with the Record type. Field names are always enclosed in double brackets, for example [[Value]].

https://tc39.es/ecma262/#sec-list-and-record-specification-type

Yes. We return an tuple (key, value) and this is more idiomatic rust (since it matches with the standards HashMap). In any case whether we return a struct or a tuple this should not matter since this is just implementation detail.